berkdb/env/env_recover.c

```diff
@@ -234,8 +234,7 @@ __fileid_track_free(dbenv)
 				ret = __db_close(r->dbp, NULL, 0);
 
 				if (ret) {
-					logmsg(LOGMSG_ERROR, "__db_close %s rc %d\n",
-						r->dbp->fname, ret);
+					logmsg(LOGMSG_ERROR, "__db_close rc %d\n", ret);
 					return ret;
```

Comment:
Might as well continue here.

Reply:
I would close the db, print `r->fname` on error, and finally free `r->fname`.

Reply:
Yea, good idea.